### DISCUSSION TITLE
New components not appearing on karavan component pallet

### DISCUSSION THREAD

**Discussion by VandanaSinha** [apache/camel-karavan#786] [Q&A] [answered]
I took camel and karavan both repository in my local and I kept my test component in the camel/components at same place where other components are and ran the build. i can see my component is listed in component.properties and its json is also included in component.json. Now i built the karavan app as per the documentation(https://github.com/apache/camel-karavan/blob/main/DEV.md
![component catalogue](https://github.com/apache/camel-karavan/assets/129922932/c498f2ff-80d0-4cc2-94ca-ae444a499408)
![no test component](https://github.com/apache/camel-karavan/assets/129922932/70ad76bf-4694-4879-b980-dfeee2713793)
). But can't see the test component in the karavan component Pallette. Please help me what should be exact step to get my component listed in karavan component Pallette.

What more i have tried is , I generate the code then pasted the code in the installed folder of karavan vsCode extension but that also didn't work, it shows the component in component catalogue but not on the actual component Pallette which appears after we click on create integration. screenshot attached.

**Comment by mgubaidullin** [comment]
1. Current version of Karavan works with Camel 3.20+
2. Current version of Karavan works with Components from Camel project.

However Karavan supports custom Kamelets, so, as a workaround I would recommend to create a custom Kamelet.

**Comment by VandanaSinha** [comment]
Thanks. 
1. I am using camel version 3.20.1. 
2. Please correct my understanding. I took code for camel, i took code for karavan in my local. build camel code, generated new properties and json file for camel catalogue. now i built the karavan and those updated files got copied into karavan app and designer. So will it still take Apache camel catalogue or will it refer my files now. Which class fetches catalogue information? I want to test my component first locally with its look and feel in karavan integration before submitting to apache repository.

**Comment by mgubaidullin** [comment, marked as answer]
karavan-generator uses camel-catalog jar to generate component list. 
you could manually add your component metadata json to components.json created by karavan-generator

**Reply by 1310208262** [reply]
how to that? add custom component  to  karavan

**Comment by VandanaSinha** [comment]
yes sir, i did the same as well as i ran the build and got those files copied. So now i can see the components in the component catalogue but not on the pallet of create integration.

Then i tried to run the karavan UI and backend locally but was getting 404 for all endpoints. i configured "proxy":"localhost:8080" in the package.json under karavan-app/src/webui. most of the 404 was resolved but now getting error when i click on "create integration " button as below:
_command 'karavan.create-yaml' not found: Error: command 'karavan.create-yaml' not found_
This command is in karavan-vscode so i think click event should go to port 3000 but due to proxy it is landing on 8080. if i configure conditional proxy i get error as below:
_When specified, "proxy" in package.json must be a string.
Instead, the type of "proxy" was "object".
Either remove "proxy" from package.json, or make it a string._
So please guide me if my journey till here is correct or we can do it in some other way. if it's right, then can you please guide me how to resolve above error.

**Comment by davsclaus** [comment]
I wonder if your camel-arrow component can be made public and donated to ASF Camel, so it can be included out of the box for everyone to use?
